tests/e2e/vsphere_file_volume_basic_mount.go

```diff
@@ -144,6 +144,140 @@ var _ = ginkgo.Describe("[csi-file-vanilla] Verify Two Pods can read write files
+
+			1. Create StorageClass without specifying fsType
+			2. Create a PVC1 with "ReadWriteMany" using the SC from above
+			3. Wait for PVC1 to be Bound
+			4. Get the VolumeID from PV
+			5. Verify using CNS Query API if VolumeID retrieved from PV is present. Also verify
+				Name, Capacity, VolumeType, Health matches
+			6. Verify if VolumeID is created on one of the VSAN datastores from list of datacenters provided in vsphere.conf
+			7. Create Pod1 using PVC1 created above at a mount path specified in PodSpec
+			8. Create a file (file1.txt) at the mount path. Check if the creation is successful
+			9. Delete the Pod and create a PVC2 with same file share created with PVC1
+			10. Create Pod2 with PVC created above, Wait for Pod2 to be Running
+			11. Read the file (file1.txt) created in Step 8 from Pod2. Check if reading is successful
+			12. Create a new file (file2.txt) at the mount path from Pod2. Check if the creation is successful
+		Cleanup:
+			1. Delete all the Pods, pvcs and storage class and verify the deletion
+	*/
+	ginkgo.It("[csi-file-vanilla] Verify File Volume is created without specifing fstype in pv spec", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// Create Storage class and PVC
+		ginkgo.By("Creating Storage Class and PVC With nfs4")
+		var storageclass *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var err error
+		storageclass, pvclaim, err = createPVCAndStorageClass(client, namespace, nil, nil, "", nil, "", false, accessMode)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		// Waiting for PVC to be bound
+		var pvclaims []*v1.PersistentVolumeClaim
+		pvclaims = append(pvclaims, pvclaim)
+		ginkgo.By("Waiting for all claims to be in bound state")
+		persistentvolumes, err := fpv.WaitForPVClaimBoundPhase(client, pvclaims, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		volHandle := persistentvolumes[0].Spec.CSI.VolumeHandle
+
+		//clean up for pvc
+		defer func() {
+			err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		// Verify variuos properties Capacity, VolumeType, datastore and datacenter of volume using CNS Query API
+		verifyVolPropertiesFromCnsQueryResults(e2eVSphere, volHandle)
+
+		//Create Pod1 with pvc created above
+		ginkgo.By("Create Pod1 with pvc created above")
+		pod1, err := fpod.CreatePod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		//cleanup for Pod1
+		defer func() {
```

Comment:
I see you have `derfer` for the deletion of SC, PVC, POD. Can we add this to our test as `after each` block or in the test itself as cleanup tasks?

if you defer all of these, you do not have control over the sequence for example - You want to delete the pod first, then PVCs, and then storageclass. you may hit into unknown issues.

Reply:
I just read about `defer` -> https://tour.golang.org/flowcontrol/13

>  Deferred function calls are pushed onto a stack. When a function returns, its deferred calls are executed in last-in-first-out order. 

we are good, you can ignore this comment.